output/output.cpp

```diff
@@ -60,7 +60,16 @@ void Output::OutputReady(void *mem, size_t size, int64_t timestamp_us, bool keyf
 
 	// Save timestamps to a file, if that was requested.
 	if (fp_timestamps_)
-		fprintf(fp_timestamps_, "%" PRId64 ".%03" PRId64 "\n", last_timestamp_ / 1000, last_timestamp_ % 1000);
+	{
+		timestampReady();
+	}
+}
+
+void Output::timestampReady()
+{
+	fprintf(fp_timestamps_, "%" PRId64 ".%03" PRId64 "\n", last_timestamp_ / 1000, last_timestamp_ % 1000);
+	if (options_->flush)
+		fflush(fp_timestamps_);
 }
```

Comment:
What do you think about passing a timestamp value as a parameter into Output::timestampReady()?

This will allow us CircularOutput::~CircularOutput() to call into Output::timestampReady() with the timestamp values passed in, and avoid duplicating this code.